webapp/channels/src/components/dropdown_input.tsx

```diff
@@ -99,30 +99,29 @@ const DropdownInput = <T extends ValueType>(props: Props<T>) => {
     const [customInputLabel, setCustomInputLabel] = useState<CustomMessageInputType>(null);
     const ownValue = useRef<T>();
 
-    const ownOnChange = (value: T, action: ActionMeta<T>) => {
+    const ownOnChange = useCallback((value: T, action: ActionMeta<T>) => {
         ownValue.current = value;
         onChange(value, action);
-    };
-    const validateInput = () => {
+    }, [onChange]);
+
+    const validateInput = useCallback(() => {
         if (!props.required || (ownValue.current !== null && ownValue.current)) {
             setCustomInputLabel(null);
             return;
         }
 
         const validationErrorMsg = formatMessage({id: 'widget.input.required', defaultMessage: 'This field is required'});
         setCustomInputLabel({type: ItemStatus.ERROR, value: validationErrorMsg});
-    };
-
-    const onInputBlur = (event: React.FocusEvent<HTMLElement>) => {
-        const {onBlur} = props;
+    }, [props.required, formatMessage]);
```

Comment:
Out of Scope:

I feel weird about mixing decomposing the props on line 84 and using those as in line 105, while here or line 108 using the props as `props.something`. It would be great to have some coherence in the file.

I think maybe the reason is because some of these are part of `otherProps`, but still, in that case I would prefer referencing `otherProps.something` instead of `props.something`.